libs/barista-components/table/src/order/order-cell.ts

```diff
@@ -0,0 +1,158 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  ViewEncapsulation,
+  OnChanges,
+  SimpleChanges,
+  ChangeDetectorRef,
+  ElementRef,
+  AfterViewInit,
+  Optional,
+  Host,
+  ViewChild,
+  OnDestroy,
+} from '@angular/core';
+import { FormControl, Validators } from '@angular/forms';
+import { DtOrder } from './order';
+import { DtCell, DtColumnDef } from '../cell';
+import { DtIndicator, _addCssClass } from '@dynatrace/barista-components/core';
+import { ENTER } from '@angular/cdk/keycodes';
+import { Subject } from 'rxjs';
+import { takeUntil, startWith } from 'rxjs/operators';
+
+// Pattern used for validating the input field
+const INPUT_PATTERN = '[0-9]*';
+
+/**
+ * Cell template that adds the right classes, role and static content for the order cell,
+ * which can be used to change the order of rows in a table.
+ */
+@Component({
+  selector: 'dt-order-cell',
+  templateUrl: './order-cell.html',
+  styleUrls: ['./order-cell.scss'],
+  host: {
+    class: 'dt-order-cell',
+    role: 'gridcell',
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.Emulated,
+  preserveWhitespaces: false,
+  exportAs: 'dtOrderCell',
+})
+export class DtOrderCell<T> extends DtCell
+  implements AfterViewInit, OnChanges, OnDestroy {
+  /** Index of the row to display in the input field */
+  @Input() index: string;
+
+  /**
+   * @internal Used to trigger the animation of the indicator to highlight
+   * a row after an order change so that it is easier for the user to follow
+   * the row to the new position
+   */
+  _animateOrderChangedIndicator = false;
+
+  /** Previous index of the row to pass to the order function */
+  private _prevIndex: number;
+
+  /** Subject used for unsubscribing */
+  private _destroy$ = new Subject<void>();
+
+  @ViewChild('orderInput', { static: true }) _orderInput: ElementRef<
+    HTMLInputElement
+  >;
+
+  _orderFormControl = new FormControl('', [
+    Validators.required,
+    Validators.pattern(INPUT_PATTERN),
+  ]);
+
+  constructor(
+    public _columnDef: DtColumnDef,
+    public _changeDetectorRef: ChangeDetectorRef,
```

Comment:
Does not need to be public
```suggestion
    columnDef: DtColumnDef,
    changeDetectorRef: ChangeDetectorRef,
```